tket/src/Predicates/PassGenerators.cpp

```diff
@@ -266,7 +266,13 @@ PassPtr gen_cx_mapping_pass(
       gen_rebase_pass(gate_set, CircPool::CX(), CircPool::tk1_to_tk1);
   PassPtr return_pass =
       rebase_pass >> gen_full_mapping_pass(arc, placement_ptr, config);
-  if (delay_measures) return_pass = return_pass >> DelayMeasures();
+  if (delay_measures) {
+    PassPtr delay_meas = WrapError<CircuitInvalidity>(
+        DelayMeasures(),
+        "`delay_measures` is set, but the circuit contains measurements that "
+        "cannot be delayed.");
+    return_pass = return_pass >> delay_meas;
+  }
```

Comment:
Can we add a test demonstrating that this message is part of the exception in this case?